processor/parse/flavored/callout.js

```diff
@@ -1,9 +1,16 @@
-// @note: We'd like to allow any emoji to match, but included in the emoji
-// character set is [#*0-9].
-// https://www.unicode.org/Public/13.0.0/ucd/emoji/emoji-data.txt
-const rgx =
-  // eslint-disable-next-line unicorn/no-unsafe-regex
-  /^> ?(\u00a9|\u00ae|[\u2000-\u3300]|[\u{1f000}-\u{1fbff}])(?: {0,}(.+))?\n((?:>(?: .*)?\n)*)/u;
+const emojiRegex = require('emoji-regex');
+
+// We're pulling out the necessary regex from `emoji-regex` because it ships itself with regex
+// slashes and if we were to embed `emoji-regex` directly into our regex for matching callouts the
+// callout regex wouldn't work at all.
+let ergx = emojiRegex().toString();
+ergx = ergx.substring(1, ergx.length - 2);
+
+// `emoji-regex` can't match unicode emoji like `✎` that folks might use as callout icons so we
+// need to augment our regex a bit.
+const unicodeRgx = '\u00a9|\u00ae|[\u2000-\u3300]|[\u{1f000}-\u{1fbff}]';
+
+const rgx = new RegExp(`^> ?(${unicodeRgx}|${ergx})(?: {0,}(.+))?\n((?:>(?: .*)?\n)*)`, 'u');
```

Comment:
> _The order of `unicodeRegex` and `ergx` in here is also curious because sometimes ℹ️ gets matched but then is translated into ⓘ in tests._

Yeah, I've run in to this issue before. For some reason, that specific emoji is *extremely* finicky!

Reply:
I think it's that we're detecting ℹ️ and then converting it to ⓘ. I don't know enough about this work to understand why though.

Reply:
WTF?! https://www.unicode.org/Public/emoji/15.0/emoji-test.txt

Reply:
How are ✎ and ✐ not emojis?

Reply:
They're just standard unicode glyphs, I think. They long pre-date emoji!

Reply:
Which appear to have all been added at the same time: https://en.wikipedia.org/wiki/Dingbat

Reply:
maybe worth submitting an issue? https://github.com/node-unicode/node-unicode-data/issues

Reply:
https://superuser.com/a/1602606/159819

> Here is a direct link for the most recent [version 13.1](https://www.unicode.org/Public/emoji/13.1/emoji-sequences.txt) from September 2020 and you can compare and contrast with [version 1.0](https://www.unicode.org/Public/emoji/1.0/emoji-data.txt) from August 2015. Lots have changed and continues to change; to code against emoji means you will be coding against a moving target. That’s sadly the nature of this beast.

Sorry, I'll stop spamming notifications.

Reply:
Ok last one I swear:

https://www.unicode.org/faq/emoji_submission.html

> Q: Can existing pictographic characters be emojified?
> No. Such proposals are no longer accepted. Proposals for emojifying existing pictographic characters were accepted in the past, but that proved to be problematic for a variety of reasons.

Reply:
I don't know what the ramifications are exactly, are we going to have to back pedal when we find out a bunch of users are using non-emoji emojis?

Otherwise, yes, 100%.

Reply:
I think we can rip them out, I went and looked at some feedback, and I don't see anyone using any non-emoji pictographs.

Reply:
dope, lmk if you need any help

Reply:
FTR, part of the problem was the `'u'` flag.

IIUC, any of the emojis from the higher code points, ones that require 32-bits to represent their code point, like 📘 (`\u{1F4D8}`), have a surrogate pair representation, `\uD83D\uDCD8`, like we use in our code. This is also how the `emoji-regex` regex tries to match them.

I suspect, setting the `'u'` flag will then only match the full code point `\u{1F4D8}`, and not the surrogate pair `\uD83D\uDCD8`?!